cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -341,6 +341,11 @@ public void writeProperties(FullHttpRequest request, HttpResponder responder,
       throw new BadRequestException("Unable to read properties from the request.", e);
     }
 
+    if (properties == null) {
+      throw new BadRequestException("Properties attribute is missing in the json or set as null, " +
+                                      "add properties: {} in json");
```

Comment:
The error message is unclear. This API expects the request body is a properties map, not an attribute inside. Maybe reword it as: `"Missing properties from the request. Please check that the request body is a json map from string to string."`

Reply:
done